apps/rewards/app/components/App/App.js

```diff
@@ -111,6 +125,25 @@ class App extends React.Component {
     })
   }
 
+  updatePath = (props) => {
+    const { selected, rewardId } = props.fromPath
+    this.state.selected = selected
```

Comment:
The state should never be directly assigned as this, this is forbidden by react rules. Apart from that, I don't think we need to use state, when we can just rely on the prop directly.

Reply:
We can instead use defaultProps if we need a default value? I don't really understand if that is what you were trying here.